streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCache.java

```diff
@@ -184,6 +184,11 @@ public synchronized V delete(K key) {
         throw new UnsupportedOperationException("MemoryLRUCache does not support all() function.");
     }
 
+    @Override
+    public KeyValueIterator<K, V> prefixScan(K prefix) {
+        throw new UnsupportedOperationException(getClass().getSimpleName() + " can't perform a prefix scan");
```

Comment:
this violates LSP, this is not a proper subclass for ReadOnlyKeyValueStore. Better to create a new interface that contains `KeyValueIterator<K, V> prefixScan(K prefix);` and then types (key value stores) that actually support this operation implement this interface.